src/main/java/randoop/contract/ObserverEqArray.java

```diff
@@ -0,0 +1,217 @@
+import randoop.sequence.Value;
+import randoop.types.JavaTypes;
+import randoop.types.PrimitiveTypes;
+import randoop.types.TypeTuple;
+
+/**
+ * A check recording the current state of an array during execution.
+ *
+ * <p>ObserverEqArray checks are not checks that must hold of all objects of a given class (unlike a
+ * check like {@link EqualsReflexive}, which must hold for any objects, no matter its execution
+ * context).
+ */
+public final class ObserverEqArray extends ObjectContract {
+
+  /** The run-time result of calling the observer: an array of literals. */
+  public Object value;
+
+  /** The maximum length (inclusive) of arrays in generated tests. */
+  public static final int MAX_ARRAY_LENGTH = 25;
+
+  /**
+   * The maximum delta between the expected and actual for which both numbers (doubles or floats)
+   * are still considered equal.
+   */
+  public static final double FLOATING_POINT_DELTA = 1e-15;
+
+  /**
+   * Create a new ObserverEqArray.
+   *
+   * @param value an array that is the run-time result of calling the observer
+   */
+  public ObserverEqArray(Object value) {
+    this.value = value;
+    if (!isLiteralType(value.getClass().getComponentType())) {
+      throw new RandoopBug(
+          String.format(
+              "Cannot represent %s as a literal",
+              StringsPlume.toStringAndClass(value.getClass().getComponentType())));
+    }
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (o == this) {
+      return true;
+    }
+    if (!(o instanceof ObserverEqArray)) {
+      return false;
+    }
+    ObserverEqArray other = (ObserverEqArray) o;
+    return Arrays.equals(toObjectArray(value), toObjectArray(other.value));
+  }
+
+  @Override
+  public int hashCode() {
+    return Arrays.hashCode(toObjectArray(value));
+  }
+
+  /**
+   * Returns true if values of the class can be represented as literals in Java source code.
+   *
+   * @param cls a class
+   * @return true iff the class is a primitive, boxed primitive, String, Class, or Enum
+   */
+  private boolean isLiteralType(Class<?> cls) {
+    if (cls == Class.class || cls == String.class || cls.isEnum()) {
+      return true;
+    }
+    return PrimitiveTypes.isBoxedPrimitive(cls) || cls.isPrimitive();
+  }
+
+  @Override
+  public String toCodeString() {
+    StringBuilder b = new StringBuilder();
+    if (Array.getLength(value) < MAX_ARRAY_LENGTH) {
+      if (value.getClass().getComponentType() == float.class
+          || value.getClass().getComponentType() == double.class) {
+        b.append(
+            String.format(
+                "org.junit.Assert.assertArrayEquals(x0, %s, %s);",
+                printArray(), FLOATING_POINT_DELTA));
+      } else {
+        b.append(String.format("org.junit.Assert.assertArrayEquals(x0, %s);", printArray()));
+      }
+    }
+    return b.toString();
+  }
+
+  /**
+   * Prints the code string of the second parameter (instantiation of array) in assertArrayEquals
```

Comment:
End sentences with punctuation.  (But don't end `@param` and `@return` clauses -- which are not full sentences -- with a period unless a full sentence follows.)